### DISCUSSION TITLE
Computed Option Name for BelongsToSelect Field

### DISCUSSION THREAD

**Discussion by intrepidws** [filamentphp/filament#1404] [Help] [answered]
I am using the following code to create a BelongsToSelect field:

```
Forms\Components\BelongsToSelect::make('discount_id')
    ->relationship('discount', 'code'),
```

Ideally, I'd like to have the label of the available options to show output from the concatenation of the `code` and `title` fields, instead of just showing the `code` field. I could add a generated MySQL field but that seems like overkill - is this perhaps possible via a closure of some kind?

**Comment by danharrin** [comment, marked as answer]
You'd need to just use a normal Select with custom `->options()` where you can define whatever key/values you wish.

**Reply by intrepidws** [reply]
Thanks Dan. Super helpful.
